### PR TITLE
Remove ES internal deletion policies in favour of Lucenes implementations

### PR DESCRIPTION
**Pull request by s1monw**
These classes are really duplicates and are just here for historical reasons.
We don't need these anymore since the same classes exist in lucene today.
This also removes the guice injection for DeletionPolicy and make them shard private.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/index/engine/ShadowEngineTests.java [hunk 1]
```diff
@@ -185,15 +183,10 @@ public long throttleTimeInNanos() {
         return new Store(shardId, EMPTY_SETTINGS, directoryService, new DummyShardLock(shardId));
     }
 
-    protected IndexDeletionPolicy createIndexDeletionPolicy() {
-        return new KeepOnlyLastDeletionPolicy(shardId, EMPTY_SETTINGS);
-    }
-
     protected SnapshotDeletionPolicy createSnapshotDeletionPolicy() {
-        return new SnapshotDeletionPolicy(createIndexDeletionPolicy());
+        return new org.apache.lucene.index.SnapshotDeletionPolicy(new KeepOnlyLastCommitDeletionPolicy());
```
Comment (mikemccand) [inline core/src/test/java/org/elasticsearch/index/engine/ShadowEngineTests.java]: This need not be fully qualified right?  (Remove the org.apache.lucene.index. prefix)

**Comment by mikemccand** [comment]
Wow :)  LGTM, I just left one nano comment.

**Comment by jpountz** [comment]
LGTM!

File: core/src/test/java/org/elasticsearch/index/engine/ShadowEngineTests.java (hunk 1 above)
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/index/engine/ShadowEngineTests.java]: oh leftover... I will fix

**Pull request merged**